examples/snmp/apc_ups_temperature_humidity.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * This script can monitor the UPS Temperatury and Humidity.
+ * Communication protocol is SNMP V2
+ * Creates a table with the following columns:
+ *      - Probe Index: The index of the prob
+ *      - Room Temperature: The current temperature reading from the probe
+ *      - Room Humidity: The current humidity reading from the probe in percent
+ */
+
+var iemStatusProbesTableOID = "1.3.6.1.4.1.318.1.1.10.2.3.2";
+var probeIndexOID = "1.3.6.1.4.1.318.1.1.10.2.3.2.1.1.1";
+var probeTempOID = "1.3.6.1.4.1.318.1.1.10.2.3.2.1.4.1";
+var probeTempUnitsOID = "1.3.6.1.4.1.318.1.1.10.2.3.2.1.5.1";
+var probeHumidityOID = "1.3.6.1.4.1.318.1.1.10.2.3.2.1.6.1";
+
+function validateAndGetData() {
+    var d = D.q.defer();
+    D.device.createSNMPSession().walk(iemStatusProbesTableOID, function(out, err) {
+        if (err) {
+            console.error(err);
+            d.failure(D.errorType.GENERIC_ERROR);
+        } else if (!out) {
+            d.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else {
+            d.resolve(out);
+        }
+    });
+
+    return d.promise;
+}
+
+/**
+ * @remote_procedure
+ * @label Validate Association
+ * @documentation This procedure is used to validate if the driver can be applied on a device during association as well as validate any credentials provided
+ */
+function validate(){
+    validateAndGetData()
+        .then(function(){
```

Comment:
This will generally success and does not really validate the association.

You must make sure the OIDs needed are part of the response data